plugins/pre-commit.sh

```diff
@@ -29,38 +29,34 @@ fi
 ########## PIP VERSION #############
 # Verify that yapf is installed; if not, warn and exit.
 if ! command -v yapf >/dev/null; then
-  echo 'yapf not on path; can not format. Please install yapf:'
-  echo '    pip install yapf'
+  printf 'yapf not on path; can not format. Please install yapf:\n' \
+    '    pip install yapf'
   exit 2
 fi
 ######### END PIP VERSION ##########
 
 ########## PIPENV VERSION ##########
 # if ! pipenv run yapf --version 2>/dev/null 2>&1; then
-#   echo 'yapf not on path; can not format. Please install yapf:'
-#   echo '    pipenv install yapf'
+#   printf 'yapf not on path; can not format. Please install yapf:\n' \
+#     '    pipenv install yapf'
 #   exit 2
 # fi
 ###### END PIPENV VERSION ##########
 
 
 # Check for unstaged changes to files in the index.
 CHANGED_FILES=$(git diff --name-only "${PYTHON_FILES[@]}")
-if [ "${CHANGED_FILES}" ]; then
-  echo 'You have unstaged changes to some files in your commit; skipping '
-  echo 'auto-format. Please stage, stash, or revert these changes. You may '
-  echo 'find `git stash -k` helpful here.'
-  echo
-  echo 'Files with unstaged changes:'
-  for file in "${CHANGED_FILES[@]}"; do
-    printf '  %s\n' "${CHANGED_FILES[@]}"
-  done
+if [ -n "${CHANGED_FILES}" ]; then
+  printf 'You have unstaged changes to some files in your commit; skipping ' \
```

Comment:
this won't print correctly.  you should leave the `echo` statements alone and only convert the for loop to a single printf call.